[BUG] version specifier doesn't work with scoped packages in `npm init`

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When using `npm init` the version specifier doesn't work when the package is scoped.



### Expected Behavior

The version specifier is used and the command doesn't error.

### Steps To Reproduce

```
$ npm create @angular@next
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "@angular" of package "@angular@next/create": name can only contain URL-friendly characters.
```

### Environment

- npm: 8.11.0
- Node.js: 16.15.1
- OS Name: Linux 5.17 Debian GNU/Linux rodete 12 (rodete)
- System Model Name: cloud

That doesn't works either

```
npm init @angular/create@next
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@angular%2fcreate-create - Not found
npm ERR! 404 
npm ERR! 404  '@angular/create-create@next' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

```

--- a/docs/content/commands/npm-init.md
+++ b/docs/content/commands/npm-init.md
@@ -38,6 +38,8 @@ follows:
 * `npm init foo` -> `npm exec create-foo`
 * `npm init @usr/foo` -> `npm exec @usr/create-foo`
 * `npm init @usr` -> `npm exec @usr/create`
+* `npm init @usr@2.0.0` -> `npm exec @usr/create@2.0.0`
+* `npm init @usr/foo@2.0.0` -> `npm exec @usr/create-foo@2.0.0`
 
 If the initializer is omitted (by just calling `npm init`), init will fall
 back to legacy init behavior. It will ask you a bunch of questions, and

--- a/lib/commands/init.js
+++ b/lib/commands/init.js
@@ -85,8 +85,13 @@ class Init extends BaseCommand {
     const [initerName, ...otherArgs] = args
     let packageName = initerName
 
+    // Only a scope, possibly with a version
     if (/^@[^/]+$/.test(initerName)) {
-      packageName = initerName + '/create'
+      const [, scope, version] = initerName.split('@')
+      packageName = `@${scope}/create`
+      if (version) {
+        packageName = `${packageName}@${version}`
+      }
     } else {
       const req = npa(initerName)
       if (req.type === 'git' && req.hosted) {

--- a/test/lib/commands/init.js
+++ b/test/lib/commands/init.js
@@ -136,6 +136,44 @@ t.test('npm init @scope/name', async t => {
   await init.exec(['@npmcli/something'])
 })
 
+t.test('npm init @scope@spec', async t => {
+  t.plan(1)
+  npm.localPrefix = t.testdir({})
+
+  const Init = t.mock('../../../lib/commands/init.js', {
+    libnpmexec: ({ args }) => {
+      t.same(
+        args,
+        ['@npmcli/create@foo'],
+        'should npx with scoped packages'
+      )
+    },
+  })
+  const init = new Init(npm)
+
+  process.chdir(npm.localPrefix)
+  await init.exec(['@npmcli@foo'])
+})
+
+t.test('npm init @scope/name@spec', async t => {
+  t.plan(1)
+  npm.localPrefix = t.testdir({})
+
+  const Init = t.mock('../../../lib/commands/init.js', {
+    libnpmexec: ({ args }) => {
+      t.same(
+        args,
+        ['@npmcli/create-something@foo'],
+        'should npx with scoped packages'
+      )
+    },
+  })
+  const init = new Init(npm)
+
+  process.chdir(npm.localPrefix)
+  await init.exec(['@npmcli/something@foo'])
+})
+
 t.test('npm init git spec', async t => {
   t.plan(1)
   npm.localPrefix = t.testdir({})

